### PR TITLE
Add details to HTTP 401 and 403 errors

### PR DESCRIPTION
**Pull request by t1agob**
**Context:** 
When getting a HTTP 401 or 403 errors, the messages are not very clear and are over simplified. 

This PR aims to solve that by adding more information to HTTP 401 and 403 error messages. 

**What is included in the PR**
- New logic that differentiates between different errors when HTTP status code is 403
- More descriptive error message on HTTP 401
- New tests to validate new logic

### CODE REVIEW

**Review by michdim — APPROVED** [review]
Great job Tiago! I added a few comments but feel free to merge as is if you disagree. Thanks!

File: validatetoken.go [hunk 1]
```diff
@@ -174,17 +188,17 @@ func (azureJwt *AzureJwtPlugin) ExtractToken(request *http.Request) (*AzureJwt,
 	authHeader, ok := request.Header["Authorization"]
 	if !ok {
 		fmt.Println("No header token")
-		return nil, errors.New("no token")
+		return nil, errors.New("no header token")
```
[the remark this answers is not in the document]
Comment (t1agob) [reply validatetoken.go]: updated and added to the PR

File: validatetoken_test.go [hunk 2]
```diff
@@ -75,6 +75,55 @@ func TestExpiredToken(t *testing.T) {
...
+func TestMissingAuthorizationHeaderToken(t *testing.T) {
+	azureJwtPlugin := AzureJwtPlugin{
+		config: &Config{
+			Issuer:   "random-issuer",
+			Audience: "admin",
+			Roles:    []string{"test_role_1"},
+		},
+	}
+
+	extractedToken, err := createUnauthorizedHeaderMissingRequest(t, azureJwtPlugin)
```
[the remark this answers is not in the document]
Comment (t1agob) [reply validatetoken_test.go]: update and added to the PR.

File: validatetoken.go [hunk 3]
```diff
@@ -174,17 +188,17 @@ func (azureJwt *AzureJwtPlugin) ExtractToken(request *http.Request) (*AzureJwt,
...
 	}
 	auth := authHeader[0]
 	if !strings.HasPrefix(auth, "Bearer ") {
 		fmt.Println("No bearer token")
-		return nil, errors.New("no token")
+		return nil, errors.New("no bearer token")
```
[the remark this answers is not in the document]
Comment (t1agob) [reply validatetoken.go]: updated to "not bearer auth scheme" and added to PR

**Review by michdim — APPROVED** [review]

**Review by michdim — APPROVED** [review]

**Pull request merged**